lcd.lua

```diff
@@ -129,15 +129,29 @@ local clearscreen = function(pos)
 end
 
 local prepare_writing = function(pos)
-	local lcd_info = lcds[minetest.get_node(pos).param2]
-	if lcd_info == nil then return end
-	local text = minetest.add_entity(
-		{x = pos.x + lcd_info.delta.x,
-		 y = pos.y + lcd_info.delta.y,
-		 z = pos.z + lcd_info.delta.z}, "digilines_lcd:text")
-	text:setyaw(lcd_info.yaw or 0)
-	--* text:setpitch(lcd_info.yaw or 0)
-	return text
+	local existing
+	local objects = minetest.get_objects_inside_radius(pos, 0.5)
+	for _, o in ipairs(objects) do
+		local o_entity = o:get_luaentity()
+		if o_entity and o_entity.name == "digilines_lcd:text" then
+			existing = o_entity
+			break
+		end
+	end
+	if not existing or existing == nil then
+		local lcd_info = lcds[minetest.get_node(pos).param2]
+		if lcd_info == nil then return end
+		local text = minetest.add_entity(
+			{x = pos.x + lcd_info.delta.x,
+			y = pos.y + lcd_info.delta.y,
+			z = pos.z + lcd_info.delta.z}, "digilines_lcd:text")
+		text:setyaw(lcd_info.yaw or 0)
+		return text
+	else
+		local meta = minetest.get_meta(existing.object:getpos())
+		local text = meta:get_string("text")
+		existing.object:set_properties({textures={generate_texture(create_lines(text))}})
```

Comment:
`existing.object:getpos() == pos`?!

Couldn't you simply do `existing:on_activate()`?

For consistence you could add `return existing.object`.

Reply:
Consistency? Why? It's extra useless code.

Reply:
(Or remove the `return` in the other case if it's is also not needed there.)